> /home/travis/build/TrinityCore/TrinityCore/src/server/game/AI/CoreAI/UnitAI.h:161:41: fatal error: 
>       comparison of distinct pointer types ('Unit *' and 'HostileReference *')
>                 if ((*itr)->getTarget() != currentVictim && predicate((*...
>                     ~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~

--- a/src/server/game/AI/CoreAI/UnitAI.h
+++ b/src/server/game/AI/CoreAI/UnitAI.h
@@ -151,11 +151,11 @@ class TC_GAME_API UnitAI
                 return nullptr;
 
             std::list<Unit*> targetList;
-            HostileReference* currentVictim = me->getThreatManager().getCurrentVictim();
+            Unit* currentVictim = me->getThreatManager().getCurrentVictim()->getTarget();
 
             // Current victim always goes first
             if (predicate(currentVictim))
-                targetList.push_back(currentVictim->getTarget());
+                targetList.push_back(currentVictim);
 
             for (ThreatContainer::StorageType::const_iterator itr = threatlist.begin(); itr != threatlist.end(); ++itr)
                 if ((*itr)->getTarget() != currentVictim && predicate((*itr)->getTarget()))

